verification/cmd/main.go

```diff
@@ -2,28 +2,46 @@ package main
 
 import (
 	"log"
+	"os"
 
-	"github.com/veraison/services/config"
+	"github.com/spf13/viper"
 	"github.com/veraison/services/verification/api"
 	"github.com/veraison/services/verification/sessionmanager"
 	"github.com/veraison/services/verification/verifier"
 	"github.com/veraison/services/vtsclient"
 )
 
 var (
-	ListenAddr  = "localhost:8080"
-	VerifierCfg = config.Store{
-		// placeholder, empty for now
-	}
-	VTSClientCfg = config.Store{
-		"vts-server.addr": "dns:127.0.0.1:50051",
-	}
+	ListenAddr = "localhost:8080"
 )
 
 func main() {
+
+	VTSClientCfg := viper.New()
+	VTSClientCfg.SetDefault("vts-server.addr", "dns:127.0.0.1:50051")
+
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	VTSClientCfg.AddConfigPath(wd)
+	VTSClientCfg.SetConfigType("yaml")
+	VTSClientCfg.SetConfigName("config")
+
+	err = VTSClientCfg.ReadInConfig()
+	if _, ok := err.(viper.ConfigFileNotFoundError); ok {
```

Comment:
I think `errors.Is(err, viper.ConfigFileNotFoundError)` should be the idiomatic way to write this

Reply:
It would be, however `errors.Is` matches on text, which could be different depending on where this originates. It is more robust to much on type here.

Reply:
Elaborating on the above comment that features a couple of inaccuracies:

> however `errors.Is` matches <s>on text</s>

Technically, `errors.Is` is little more complicated (see: https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/errors/wrap.go;l=40) It matches:
- If the target `error` is Comparable and equals (`==`) to the tested `error` 
- If the tested `error` implements an `Is` method, and calling it on target returns `true`.

`viper.ConfigFileNotFoundError` does not implement `Is` so only the first check is applied. Structs are compared based on their corresponding field values. `viper.ConfigFileNotFoundError` contains only two (hidden) fields, both of which are strings: https://github.com/spf13/viper/blob/master/viper.go#L111. So this _effectively_ amounts to matching on text.


>  which could be different depending on where <s>this originates</s>.

The problem isn't so much with where the error originates but with the two fields that get populated. They are the name of the config file, and the locations that are searched for it. Matching the string containing the locations, in particular, is going to be very brittle.

The upshot is that checking based on type seems more robust here.

Reply:
Ah, I see! Thanks for digging into this.  Would the snippet below work instead?

```go
var cfgFileNotFound viper.ConfigFileNotFoundError

if errors.As(err, &cfgFileNotFound) {
    // ...
}
```

Reply:
Yup, that works! Updated.